R/model.R

```diff
@@ -357,17 +343,15 @@ CmdStanModel <- R6::R6Class(
 #'
 #' }
 #'
-NULL
-
-compile_method <- function(quiet = TRUE,
-                           dir = NULL,
-                           pedantic = FALSE,
-                           include_paths = NULL,
-                           cpp_options = list(),
-                           stanc_options = list(),
-                           force_recompile = FALSE,
-                           #deprecated
-                           threads = FALSE) {
+compile <- function(quiet = TRUE,
```

Comment:
Taking the compile method as an example (but this works with all the other methods too), to fix the Usage section problem we can switch to defining a function `compile` (instead of `compile_method`) and we create doc for that `compile` function (instead of using `NULL` at the end of the docstring). But then we do _not_ export the `compile` function and we do _not_ use `@noRd`. This results in roxygen2 creating the Rd file as if the function were exported but it's not actually exported. We then use `CmdStanModel$set("public", name = "compile", value = compile)`. 

This also has the benefit that `sample_mpi` can use `inheritParams` to inherit from `sample` instead of having to move to roxygen templates.

Reply:
I still need to do this for all the CmdStanFit methods, but it seems to be working for all the CmdStanModel methods!